### ISSUE TITLE
Remaining GetAllSecrets implementations in community providers

### ISSUE THREAD

**Issue by knelasevero** [external-secrets/external-secrets#1110]
For Reference:

https://github.com/external-secrets/external-secrets/issues/759

https://github.com/external-secrets/external-secrets/issues/758

https://github.com/external-secrets/external-secrets/issues/757

https://github.com/external-secrets/external-secrets/issues/756

https://github.com/external-secrets/external-secrets/issues/498

☝️ Those are all done and you can look at their implementations to see how they did it.

@haf-tech @nitzann @IdanAdar 

@AndreyZamyslov

@Jabray5

@ElsaChelala

@KianTigger @EladGabay

@renanaAkeyless

@willemm

Tagging all contributors that added a provider or seemed to have corresponding infra to test it.

Would love to see those GetAllSecrets methods implemented for all providers :).

I think it makes sense to mention that not all providers can implement find by tags, so we can have "not supported" errors on those. Find by regex we can implement on our side, and I think that this would apply to all providers.

Let us know if you start working on one of those so we don't have people picking up the same work in parallel.

Also, to anyone reading this, feel free to grab any of these even if you are not listed, if you have a way to test it.

- [ ] IBM
- [ ] Yandex
- [ ] Gitlab
- [ ] Alibaba
- [ ] Oracle
- [ ] Akeyless
- [ ] K8s
- [ ] Generic Webhook

**Comment by gusfcarvalho** [comment]
I'm not sure if GetAllSecrets would be supported for Generic Webhook, would it?

**Comment by willemm** [comment]
> I'm not sure if GetAllSecrets would be supported for Generic Webhook, would it?

Not easily, no.

I imagine there could be an extension to the crd where you can define separate generic-webhook url entries for getting all secrets by tag or by regex match.  It would need to respond with 'not implemented' type errors when those aren't defined.  And probably make all of them optional so you could also have one where you only have the GetAllSecrets option?

**Comment by gusfcarvalho** [comment]
I think implementing a "Not Supported" (instead of "Not Implemented") is already enough. IMO, it is going to be the same configuration as creating multiple secret stores, so it seems a little bit pointless to me. We are not getting any functionalities.

**Comment by AndreyZamyslov** [comment]
> Yandex 

Both names and tags/labels are poorly supported by Yandex.Cloud services at the moment.
I've implemented GetAllSecrets as "Not Supported" for now: [commit](https://github.com/external-secrets/external-secrets/pull/1040/commits/f4f217050248df89122c8f5f4e9de27f850c1b5c)

**Comment by sebagomez** [comment]
FYI: I'll start digging this issue for [IBM Cloud Secrets Manager](https://www.ibm.com/cloud/secrets-manager)

**Comment by jack-evans** [comment]
@sebagomez are you still looking into this for IBM Cloud? Keen to use this feature ☺️

**Comment by sebagomez** [comment]
hey @jack-evans , no sorry, I removed myself as assigned to this issue because I couldn't work on it. Maybe in the future, but I have to plan ahead a little bit. 

I will pick it up if nobody else does it eventually.

**Comment by jack-evans** [comment]
one thing worth mentioning, the ExternalSecret expects a key value pair for tags but in the case of IBM Secrets Manager, the tags are just a string... so I could have a tag like `foobar`, `foo=bar` or `foo: bar`. May need to think about these cases to set a standard of what you expect an end user to tag their secrets with and the potential edge cases where the code _might_ panic if we dont catch them

**Comment by laisramos123** [comment]
Hi guys! I was analyzing the code and I believe that getAllSecrets from some providers in the list above is already implemented. Below is the updated list that has the getAllSecrets function implemented:

- [X] Akeyless
- [X] AWS
- [X] Azure
- [X] Doppler
- [X] GCP
- [X] GitLab
- [X] Keeper Security
- [X] Kubernetes
- [X] One Password
- [X] Scaleway

And the list of providers where the getAllSecrets function needs to be implemented:

- [ ] Alibaba
- [ ]  Conjur
- [ ] Delinea
- [ ]  Senha Segura
- [ ]  IBM
- [ ] Oracle
- [ ] Yandex

Feel free to correct me at any point ;)

**Comment by 6ixfalls** [comment]
Would it be possible to reopen the Generic Webhook GetAllSecrets implementations? While it may require a pretty large overhaul in the CRD (either adding another URL or requiring two SecretStores, one for GetAllSecrets and another for GetSecret), having the ability to implement GetAllSecrets while using webhooks is useful to add even more use cases. If GetAllSecrets were implemented on the Webhook provider, I could get rid of some secret operators designed only for one specific provider.

**Comment by moolen** [comment]
hey @6ixfalls, i appreciate your interest! Could you please open a new issue for this? I think this needs some discussion outside of this thread so we can design it properly. I don't really have a idea of how that could look like. 
I'd be interested to get your view on it and what your requirements are.

**Comment by 6ixfalls** [comment]
> hey @6ixfalls, i appreciate your interest! Could you please open a new issue for this? I think this needs some discussion outside of this thread so we can design it properly. I don't really have a idea of how that could look like. I'd be interested to get your view on it and what your requirements are.

@moolen #2896

**Comment by fdberlking** [comment]
Idk who to tag, but I'd like to take a look into getAllSecrets function for IBM's Secret Manager.
Does this work for you? :)

**Comment by gusfcarvalho** [comment]
@fdberlking the best one regarding IBM would be @IdanAdar  :)

**Comment by fdberlking** [comment]
Alright, thus I leave it up to him 👍

**Comment by matheusmazzoni** [comment]
Hi guys, does it make sense to separate this issue into for the different providers? 🤔

**Comment by anders-swanson** [comment]
Oracle Vault has had GetAllSecrets for some time: https://github.com/external-secrets/external-secrets/blob/main/pkg/provider/oracle/oracle.go#L180

**Comment by evrardj-roche** [comment]
I think we should close this and reopen it for each provider not yet implemented, wdyt? (Or alternatively, put the effort to do it)

**Comment by alekc** [comment]
Current situation 

  ```
  ┌─────┬────────────────┬──────────────────────────────┬───────────────────────────────────────────────────────────────────┐
  │  #  │    Provider    │            Status            │                               Notes                               │
  ├─────┼────────────────┼──────────────────────────────┼───────────────────────────────────────────────────────────────────┤
  │ 1   │ akeyless       │ ✅ Implemented               │ Done since #1110                                                           │
  ├─────┼────────────────┼──────────────────────────────┼────────────────────────────────────────────────────────────────────────────┤
  │ 2   │ aws            │ ✅ Implemented               │                                                                            │
  ├─────┼────────────────┼──────────────────────────────┼────────────────────────────────────────────────────────────────────────────┤
  │ 3   │ azure          │ ✅ Implemented               │                                                                            │
  ├─────┼────────────────┼──────────────────────────────┼────────────────────────────────────────────────────────────────────────────┤
  │ 4   │ barbican       │ ✅ Implemented               │                                                                            │
  ├─────┼────────────────┼──────────────────────────────┼────────────────────────────────────────────────────────────────────────────┤
  │ 5   │ beyondtrust    │ 🚫 Not implemented           │ errors.New("GetAllSecrets not implemented")                                │
  ├─────┼────────────────┼──────────────────────────────┼────────────────────────────────────────────────────────────────────────────┤
  │ 6   │ bitwarden      │ ✅ Implemented               │                                                                            │
  ├─────┼────────────────┼──────────────────────────────┼────────────────────────────────────────────────────────────────────────────┤
  │ 7   │ chef           │ 🚫 Not implemented           │ errors.New("dataFrom.find not suppported") (sic — typo in source)          │
  ├─────┼────────────────┼──────────────────────────────┼────────────────────────────────────────────────────────────────────────────┤
  │ 8   │ cloudru        │ ✅ Implemented               │                                                                            │
  ├─────┼────────────────┼──────────────────────────────┼────────────────────────────────────────────────────────────────────────────┤
  │ 9   │ conjur         │ ✅ Implemented               │                                                                            │
  ├─────┼────────────────┼──────────────────────────────┼────────────────────────────────────────────────────────────────────────────┤
  │ 10  │ delinea        │ ⛔ Not supported by upstream │ "Delinea DevOps Secrets Vault" has no list API                             │
  ├─────┼────────────────┼──────────────────────────────┼────────────────────────────────────────────────────────────────────────────┤
  │ 11  │ doppler        │ ✅ Implemented               │                                                                            │
  ├─────┼────────────────┼──────────────────────────────┼────────────────────────────────────────────────────────────────────────────┤
  │ 12  │ dvls           │ 🚫 Not implemented           │ errNotImplemented                                                          │
  ├─────┼────────────────┼──────────────────────────────┼────────────────────────────────────────────────────────────────────────────┤
  │ 13  │ fake           │ ✅ Implemented               │                                                                            │
  ├─────┼────────────────┼──────────────────────────────┼────────────────────────────────────────────────────────────────────────────┤
  │ 14  │ fortanix       │ ✅ Implemented               │                                                                            │
  ├─────┼────────────────┼──────────────────────────────┼────────────────────────────────────────────────────────────────────────────┤
  │ 15  │ gcp            │ ✅ Implemented               │                                                                            │
  ├─────┼────────────────┼──────────────────────────────┼────────────────────────────────────────────────────────────────────────────┤
  │ 16  │ github         │ ✅ Implemented               │                                                                            │
  ├─────┼────────────────┼──────────────────────────────┼────────────────────────────────────────────────────────────────────────────┤
  │ 17  │ gitlab         │ ✅ Implemented               │ Done since #1110                                                           │
  ├─────┼────────────────┼──────────────────────────────┼────────────────────────────────────────────────────────────────────────────┤
  │ 18  │ ibm            │ 🚫 Not implemented           │ errNotImplemented (// TO be implemented) — listed in #1110                 │
  ├─────┼────────────────┼──────────────────────────────┼────────────────────────────────────────────────────────────────────────────┤
  │ 19  │ infisical      │ ✅ Implemented               │                                                                            │
  ├─────┼────────────────┼──────────────────────────────┼────────────────────────────────────────────────────────────────────────────┤
  │ 20  │ keepersecurity │ ✅ Implemented               │                                                                            │
  ├─────┼────────────────┼──────────────────────────────┼────────────────────────────────────────────────────────────────────────────┤
  │ 21  │ kubernetes     │ ✅ Implemented               │ Done since #1110                                                           │
  ├─────┼────────────────┼──────────────────────────────┼────────────────────────────────────────────────────────────────────────────┤
  │ 22  │ nebius         │ 🚫 Not implemented           │ errNotImplemented                                                          │
  ├─────┼────────────────┼──────────────────────────────┼────────────────────────────────────────────────────────────────────────────┤
  │ 23  │ ngrok          │ ✅ Implemented               │                                                                            │
  ├─────┼────────────────┼──────────────────────────────┼────────────────────────────────────────────────────────────────────────────┤
  │ 24  │ onboardbase    │ 🟡 Partial                   │ Name-based find works; rejects tag-based with "find by tags not supported" │
  ├─────┼────────────────┼──────────────────────────────┼────────────────────────────────────────────────────────────────────────────┤
  │ 25  │ onepassword    │ ✅ Implemented               │                                                                            │
  ├─────┼────────────────┼──────────────────────────────┼────────────────────────────────────────────────────────────────────────────┤
  │ 26  │ onepasswordsdk │ 🚫 Not implemented           │ errNotImplemented (separate from onepassword)                              │
  ├─────┼────────────────┼──────────────────────────────┼────────────────────────────────────────────────────────────────────────────┤
  │ 27  │ oracle         │ ✅ Implemented               │ Done since #1110                                                           │
  ├─────┼────────────────┼──────────────────────────────┼────────────────────────────────────────────────────────────────────────────┤
  │ 28  │ ovh            │ ✅ Implemented               │                                                                            │
  ├─────┼────────────────┼──────────────────────────────┼────────────────────────────────────────────────────────────────────────────┤
  │ 29  │ passbolt       │ ✅ Implemented               │ Requires ref.Name.RegExp to be set                                         │
  ├─────┼────────────────┼──────────────────────────────┼────────────────────────────────────────────────────────────────────────────┤
  │ 30  │ passworddepot  │ 🚫 Not implemented           │ fmt.Errorf(errNotImplemented, "GetAllSecrets")                             │
  ├─────┼────────────────┼──────────────────────────────┼────────────────────────────────────────────────────────────────────────────┤
  │ 31  │ previder       │ 🚫 Not implemented           │ errNotImplemented                                                          │
  ├─────┼────────────────┼──────────────────────────────┼────────────────────────────────────────────────────────────────────────────┤
  │ 32  │ pulumi         │ ✅ Implemented               │                                                                            │
  ├─────┼────────────────┼──────────────────────────────┼────────────────────────────────────────────────────────────────────────────┤
  │ 33  │ scaleway       │ ✅ Implemented               │                                                                            │
  ├─────┼────────────────┼──────────────────────────────┼────────────────────────────────────────────────────────────────────────────┤
  │ 34  │ secretserver   │ ⛔ Not supported by upstream │ "Delinea Secret Server" has no list API                                    │
  ├─────┼────────────────┼──────────────────────────────┼────────────────────────────────────────────────────────────────────────────┤
  │ 35  │ senhasegura    │ 🚫 Not implemented           │ errNotImplemented                                                          │
  ├─────┼────────────────┼──────────────────────────────┼────────────────────────────────────────────────────────────────────────────┤
  │ 36  │ vault          │ ✅ Implemented               │                                                                            │
  ├─────┼────────────────┼──────────────────────────────┼────────────────────────────────────────────────────────────────────────────┤
  │ 37  │ volcengine     │ ✅ Implemented               │                                                                            │
  ├─────┼────────────────┼──────────────────────────────┼────────────────────────────────────────────────────────────────────────────┤
  │ 38  │ webhook        │ 🚫 Not implemented           │ errNotImplemented (// TO be implemented) — listed in #1110                 │
  ├─────┼────────────────┼──────────────────────────────┼────────────────────────────────────────────────────────────────────────────┤
  │ 39  │ yandex         │ 🚫 Not implemented           │ errNotImplemented (// TO be implemented) — listed in #1110                 │
  └─────┴────────────────┴──────────────────────────────┴────────────────────────────────────────────────────────────────────────────┘
```

**Comment by evrardj-roche** [comment]
How did you generate that list? Using an LLM analyzing the code? Or did you refactor the code to provide a single registry that would allow _deterministic_ implementation checking? Just curious because I did the latter somewhere.

**Comment by alekc** [comment]
Claude code Opus in loop which went through the provider code

**Comment by l00py** [comment]
Anyone working on the webhook? It's shown as picked, but it could be outdated.

**Comment by evrardj-roche** [comment]
@l00py I haven't seen a PR  and I don't see GetAllSecrets in the code, so I would say go ahead...

**Comment by calebmckay** [comment]
I ran into this issue trying to use `dataFrom` with the onepasswordSDK provider. Since it seems like most of the functionality should be the same as the onepassword (Connect Server) provider, anyone mind if I take a crack at it?

I'd probably also need to coordinate with @Skarlso and their work on 1Password Environments in https://github.com/external-secrets/external-secrets/pull/6318.

**Comment by Skarlso** [comment]
Sure, go for it.

**Comment by Abderraoufzekkour** [comment]
Hi! I'd like to implement GetAllSecrets for the webhook provider.
I can see from the codebase it currently returns errNotImplemented.
I'll use the existing provider implementations as reference.
Can I be assigned for the webhook provider?
